perfkitbenchmarker/openstack/os_network.py

```diff
@@ -64,3 +66,40 @@ def AllowPort(self, vm, from_port, to_port=None, protocol='tcp'):
 
     def DisallowAllPorts(self):
         pass
+
+
+class OpenStackPublicNetwork(object):
+
+    def __init__(self, pool):
+        self.__nclient = utils.NovaClient()
+        self.__floating_ip_lock = threading.Lock()
+        self.pool = pool
+
+    def allocate(self):
+        with self.__floating_ip_lock:
+            return self.__nclient.floating_ips.create(pool=self.pool)
+
+    def release(self, floating_ip):
+        f_id = floating_ip.id
+        if self.__nclient.floating_ips.get(f_id).fixed_ip:
```

Comment:
NovaClient is just a wrapper of NovaClient from original python-novaclient library which set up credentials and make connection. When you look to code in openstack/utils.py you have simple redirection for all methods to original client class.